src/test/java/spoon/test/compilation/CompilationTest.java

```diff
@@ -0,0 +1,91 @@
+package spoon.test.compilation;
+
+import org.junit.Assert;
+import org.junit.Test;
+import spoon.Launcher;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtBlock;
+import spoon.reflect.code.CtReturn;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.factory.CodeFactory;
+import spoon.reflect.factory.CoreFactory;
+import spoon.reflect.factory.Factory;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+
+public class CompilationTest {
+
+	@Test
+	public void compileTest() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/resources/noclasspath/Simple.java");
+		File outputBinDirectory = new File("./target/class-simple");
+		if (!outputBinDirectory.exists()) {
+			outputBinDirectory.mkdirs();
+		}
+		launcher.setBinaryOutputDirectory(outputBinDirectory);
+		launcher.getEnvironment().setShouldCompile(true);
+		launcher.buildModel();
+
+		Factory factory = launcher.getFactory();
+		CoreFactory core = factory.Core();
+		CodeFactory code = factory.Code();
+
+		CtClass simple = factory.Class().get("Simple");
+
+		CtMethod method = core.createMethod();
+		method.addModifier(ModifierKind.PUBLIC);
+		method.setType(factory.Type().integerPrimitiveType());
+		method.setSimpleName("m");
+
+		CtBlock block = core.createBlock();
+		CtReturn aReturn = core.createReturn();
+
+		CtBinaryOperator<Object> binaryOperator = code.createBinaryOperator(
+						code.createLiteral(10),
+						code.createLiteral(32),
+						BinaryOperatorKind.PLUS);
+		aReturn.setReturnedExpression(binaryOperator);
+
+		block.addStatement(aReturn);
+		method.setBody(block);
+
+		simple.addMethod(method);
+
+		launcher.getModelBuilder().compile();
+
+		final URLClassLoader urlClassLoader = new URLClassLoader(new URL[] { outputBinDirectory.toURL() });
+
+		class MethodRunner extends Thread {
+			public Object value;
+
+			@Override
+			public void run() {
+				synchronized (this) {
+					try {
+						Class<?> aClass = urlClassLoader.loadClass("Simple");
+						Method m = aClass.getMethod("m");
+						this.value = m.invoke(aClass.newInstance());
+					} catch (Exception e) {
+						Assert.fail(e.getMessage());
+					}
+					notify();
+				}
+			}
+		}
+
+		MethodRunner methodRunner = new MethodRunner();
```

Comment:
I do not need a asynchronous call but a need a thread the change the classloader.
And this is the only way that I know to wait the end of the execution of a thread.